src/main.ts

```diff
@@ -141,40 +141,42 @@ function conflictExtensionCheck() {
 }
 
 function newVersionMessage(extensionPath: string, extension: Extension) {
```

Comment:
Instead of wrapping a promise, you may simply put `async` before `function` to promisify the result. It is also git-log-friendly.

Reply:
Does that work with the callback function? I haven't done that before.

Reply:
i.e.
```
async function example() {
    functionwithcallback((data) => {
        await data
    })
}
````

because I thought you had to do promises for that?